Merged changes from nodejs/node 10.0.0 and 11.11.0

LightWayUp opened the pull request:
* url: make urlToOptions() handle IPv6 literals
  Strip the enclosing square brackets from the parsed hostname.
  Merged from nodejs/node@fa2d43b

* url: handle quasi-WHATWG URLs in urlToOptions()
  Merged from nodejs/node@0017855

* Removed whitespace in node version field

* Updated devDependencies
  * Updated semver from `^5.3.0` to `^6.0.0`
  * Updated universal-url from `^1.0.0` to `^2.0.0`

LightWayUp committed:
url: make urlToOptions() handle IPv6 literals

Strip the enclosing square brackets from the parsed hostname.

Merged from nodejs/node@fa2d43b

LightWayUp committed:
url: handle quasi-WHATWG URLs in urlToOptions()

Merged from nodejs/node@0017855

LightWayUp committed:
Removed whitespace in node version field

Whitespace should be used when forming a comparator set, and should not appear within a comparator

LightWayUp committed:
Updated all devDependencies

- Updated semver from `^5.3.0` to `^6.0.0`
- Updated universal-url from `^1.0.0` to `^2.0.0`
- Updated whatwg-url from `^5.0.0` to `^7.0.0`

All tests passed

stevenvachon commented:
Tests are failing

LightWayUp committed:
Attempting to fix tests

- NaN can't be compared with assert.strictEqual() below Node.js 9.0.0
- Destructuring assignment syntax is not available below Node.js 6.0.0

LightWayUp committed:
Downgraded whatwg-url back to 5.0.0

- Downgraded whatwg-url back to 5.0.0, later versions require `new.target` which wasn't supported until Node.js 5.0.0. Regression: d9450ff

LightWayUp commented:
Sorry, all fixed now 😅.

stevenvachon commented:
Thanks! Feel free to require at least Node.js v8.x if it helps improve tests and dependencies

LightWayUp commented:
~~I stumbled upon this when trying to use some API while still maintaining compatibility with older Node.js versions, so I think the current Node.js >= 4 requirement is fine and worth keeping; and for dependencies, only whatwg-url can't be updated but it's only used for testing under Node.js <= 6, so that's ok 🙂.~~ ( Nvm, I didn't know that support for 6.x will end after 4/30 )

stevenvachon commented:
Node 6.x is now end-of-life, but okay.

LightWayUp commented:
> Node 6.x is now end-of-life, but okay.

Oh wow, I didn't realize that it'll go EOL at the end of April. Should I remove the support for <= 6? And what do you think about running test too with the current LTS 10.x release, should I add it to the Travis config?

LightWayUp committed:
Increased Node.js version for test runs

- Increased Node.js version for tests as 4.x reached end of life in 2018, and 6 will reach EOL within a month too
- Added Node.js 10 for running tests in CI configuration as it's LTS release

LightWayUp commented:
@stevenvachon I've gone ahead and increased the Node.js versions. Please take a look at the changes, and see if it's ok to be merged, thanks!

stevenvachon commented:
Please rebase.

Do we need the tests when the code is merely copied from Node core? Perhaps a simple assertion for an exported function would be enough, and less to maintain.

LightWayUp commented:
Rebased.

As for reducing/simplifying tests, I can if you want to. However, although changes are copied from Node core, they're from newer versions and the tests could potentially catch compatibility issues with older LTS that this is still supporting ( Node.js 8 for example, which doesn't contain these changes ).

stevenvachon commented on README.md:
```diff
@@ -19,7 +19,7 @@ const urlToOptions = require('url-to-options');
 
 const opts = urlToOptions(url);
-//-> { auth:'user:pass', port:8080, … }
+//-> { auth:'user:pass', port:8080, &#x2026; }
```
This is odd

LightWayUp replied:
Hmm.. I think my editor messed it up due to the character being a bit more special, I'll change it back

stevenvachon commented on test.js:
```diff
@@ -3,7 +3,7 @@ const assert = require('assert');
 const urlToOptions = require('./');
 
-const URL = require(semver.satisfies(process.version, '>= 6') ? 'universal-url' : 'whatwg-url').URL;
+const URL = require('universal-url').URL;
```
```js
const {URL} = require('universal-url');
```

stevenvachon commented on test.js:
```diff
@@ -3,13 +3,14 @@ const assert = require('assert');
 
 
 
 // Copied from https://github.com/nodejs/node/blob/master/test/parallel/test-whatwg-url-properties.js
```
This file no longer exists (?)

LightWayUp replied:
Ah, I found it, they renamed the file

LightWayUp committed:
Made changes as requested

- Fixed link in comment (file has been renamed in node repo)
- Slight style change

stevenvachon commented:
Thanks!
